In [1]:
#dependencies
import pandas as pd

#read file and convert into dataframe and show head
file = 'Resources/purchase_data.csv'
purchase_data_df = pd.read_csv(file)
#remove rows with missing data
purchase_data_df.count()
cleaned_data_df = purchase_data_df.dropna(how='any')
cleaned_data_df.count()
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [2]:
#use length of SN's to calculate player count
total_players = len(purchase_data_df['SN'].value_counts())

#convert this into a dataframe to display
player_count = pd.DataFrame({'Total Players': [total_players]})
player_count

Total Players
0            576

Purchasing Analysis (Total)

- Number of Unique Items
- Average Purchase Price
- Total Number of Purchases
- Total Revenue

In [3]:
#Calulations for purchasing Analysis
#length of unique items
unique_items = len(purchase_data_df['Item ID'].unique())
#mean of Price
mean_price = (purchase_data_df['Price'].mean())
#total count of transactions, unique not applicable
total_purchases = (purchase_data_df['Purchase ID'].count())
#sum of the whole price column
total_revenue = (purchase_data_df['Price'].sum())

#Display Results into DF
purchasing_analysis_df = pd.DataFrame({'Number of Unique Items': [unique_items],
                                      'Average Price': [mean_price],
                                      'Total Number of Purchases': [total_purchases],
                                       'Total Revenue': [total_revenue]})
purchasing_analysis_df

#Change format of average price and total revenue to display as currency
purchasing_analysis_df.style.format({'Average Price': '${:,.2f}',
                                    'Total Revenue': '${:,.2f}'})

Gender Demographics

- Percentage and Count of Male Players
- Percentage and Count of Female Players
- Percentage and Count of Other / Non-Disclosed

In [4]:
#Grouping Purchase_data by Gender
genders_df = purchase_data_df.groupby(['Gender'])

# we count the total number of SN's by gender
gender_count = genders_df.nunique()['SN']

#Get Percentage value by gender by dividing by total players
player_percentage = gender_count / total_players * 100

#DataFrame to display these values
gender_demographics_df = pd.DataFrame({'Percent': player_percentage, 'Count of Gender': gender_count})

# Format table by decending order, format percentage, and remove gender key
gender_demographics_df.sort_values(["Count of Gender"], ascending = False).style.format({"Percent":"{:.2f}%"})
gender_demographics_df.index.name = None

gender_demographics_df


Percent  Count of Gender
Female                 14.062500               81
Male                   84.027778              484
Other / Non-Disclosed   1.909722               11

Purchasing Analysis (Gender)

The below each broken by gender

- Purchase Count
- Average Purchase Price
- Total Purchase Value
- Average Purchase Total per Person by Gender

In [5]:
#Purchase count per gender
purchase_count = genders_df['Purchase ID'].count()

#Mean purchase price per gender
mean_purchase_price = genders_df['Price'].mean()

#Mean of total purchases by gender
mean_purchase_total = genders_df['Price'].sum()

#Average Purchase Total per person by Gender
average_purchase_per_person = mean_purchase_total / gender_count

gender_analysis = pd.DataFrame({'Purchase Count': purchase_count,
                               'Average Purchase Price': mean_purchase_price,
                               'Total Purchase Value': mean_purchase_total,
                               'Average Purchase Total per Person': average_purchase_per_person})


gender_analysis.index.name = "Gender"


gender_analysis.style.format({"Total Purchase Value":"${:,.2f}",
                                  "Average Purchase Price":"${:,.2f}",
                                  "Average Purchase Total per Person":"${:,.2f}"})


Age Demographics
 - The below each broken into bins of 4 years (i.e. <10, 10-14, 15-19, etc.)

    - Percentage of Players

    - Total Count

In [6]:
#Make Bins
age_bins = [0, 9.99, 14.99, 19.99, 24.99, 29.99, 34.99, 39.99, 500]
labels = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

#Cut values by age into bins
purchase_data_df['Age Group'] = pd.cut(purchase_data_df["Age"], age_bins, labels=labels)

#new df with new Age Group column, grouped by Age Group
age_bin_df = purchase_data_df.groupby('Age Group')
# age_bin_df.count().head(10)

#count of total players in each category
age_count_categories = age_bin_df['SN'].nunique()

#calculate percentages in each age category
age_percentage = (age_count_categories / total_players) * 100

#create new dataframe
age_demographics_df = pd.DataFrame({"Percentage of Players": age_percentage, 
                                   'Total Count': age_count_categories})

age_demographics_df.index.name = None

age_demographics_df.style.format({'Percentage of Players': "{:,.2f}%"})


Purchasing Analysis (Age)¶
- The below each broken into bins of 4 years (i.e. <10, 10-14, 15-19, etc.)

    - Purchase Count

    - Average Purchase Price

    - Total Purchase Value

    - Average Purchase Total per Person by Age Group

In [7]:
# Count purchases by age group
purchase_count_age = age_bin_df["Purchase ID"].count()

# Obtain average purchase price by age group 
mean_purchase_price_age = age_bin_df["Price"].mean()

# Calculate total purchase value by age group 
total_purchase_value = age_bin_df["Price"].sum()

# Calculate the average purchase per person in the age group 
mean_purchase_per_person_age = total_purchase_value/age_count_categories

# Create data frame with obtained values
age_demographics = pd.DataFrame({"Purchase Count": purchase_count_age,
                                 "Average Purchase Price": mean_purchase_price_age,
                                 "Total Purchase Value":total_purchase_value,
                                 "Average Purchase Total per Person": mean_purchase_per_person_age})

# Format the data frame with no index name in the corner
age_demographics.index.name = None

# Format with currency style
age_demographics.style.format({"Average Purchase Price":"${:,.2f}",
                               "Total Purchase Value":"${:,.2f}",
                               "Average Purchase Total per Person":"${:,.2f}"})


Top Spenders
- Identify the the top 5 spenders in the game by total purchase value, then list (in a table):

    - SN(screen name)

    - Purchase Count

    - Average Purchase Price

    - Total Purchase Value

In [8]:
#Group Purchase_data_df by SN's
sn_grouped = purchase_data_df.groupby('SN')

#Count total purchases by each name
purchase_count_sn = sn_grouped['Purchase ID'].count()

#Calculate mean purchase by SN
mean_purchase_price_sn = sn_grouped['Price'].mean()

#Calculate purchase total
purchase_total_sn = sn_grouped['Price'].sum()


#Calculate new df with values above
top_spenders_df = pd.DataFrame({'Purchase Count': purchase_count_sn, 
                               'Average Purchase Price': mean_purchase_price_sn,
                               'Total Purchase Value': purchase_total_sn})

# Sort in descending order to obtain top 5 spender names 
formatted_top_spenders_df = top_spenders_df.sort_values(["Total Purchase Value"], ascending=False).head()

# Format with currency style
formatted_top_spenders_df.style.format({"Average Purchase Total":"${:,.2f}",
                                 "Average Purchase Price":"${:,.2f}", 
                                 "Total Purchase Value":"${:,.2f}"})

Most Popular Items¶
- Top 5 most popular items by purchase count:

    - Item ID

    - Item Name

    - Purchase Count

    - Item Price

    - Total Purchase Value

In [39]:
#We must create a new dataframe for items and associated columns
items_df = purchase_data_df[['Item ID', 'Item Name', 'Price']]

#Group items_df by Items ID and Item Name
items_grouped = items_df.groupby(['Item ID', 'Item Name'])

#count the times an item has been purchased
item_purchases = items_grouped['Price'].count()

#calculate purchase value for each item
purchase_value = items_grouped['Price'].sum()

#Find the price of each individual item
item_price = purchase_value/item_purchases

#Create popular_items_df
popular_items_df = pd.DataFrame({'Purchase Count': item_purchases,
                                'Item Price': item_price,
                                'Total Purchase Value': purchase_value})

# Sort in descending order to get top items
popular_items_formatted = popular_items_df.sort_values(["Purchase Count"], ascending=False).head()

#Format for two decimal places and add $ sign
popular_items_formatted.style.format({"Item Price":"${:,.2f}",
                                "Total Purchase Value":"${:,.2f}"})



Most Profitable Items

- Top 5 most profitable items by total purchase value:

    - Item ID

    - Item Name

    - Purchase Count

    - Item Price

    - Total Purchase Value

In [41]:
#We can take popular_items_formatted and sort that for highest total purchase value
profitable_formatted = popular_items_formatted.sort_values(['Total Purchase Value'], ascending=False).head()

#Format for two decimal places and add $ sign
profitable_formatted.style.format({'Item Price': '${:,.2f}',
                                  'Total Purchase Value': '${:,.2f}'})

Observable Trends:

- Based off of this purchase data, the game is mostly played by males who come in at 84% of gender of purchasers
- More than 60% of players making these purchases are in the age range of 15-24. Mostly high school and early post-college age.
- The 20-24 age range has the highest total purchase value but the age range 35-39 has the highest average purchase price of 4.76 . 